File: Sample CSharp Application/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Sample_CSharp_Application
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private async void button1_Click(object sender, EventArgs e)
        {
            var result = await DevNet.QImage.Query(this.textBox1.Text, (int)this.numericUpDown1.Value);
            dataGridView1.DataSource = result;
        }

        private void dataGridView1_CellContentDoubleClick(object sender, DataGridViewCellEventArgs e)
        {
            if (dataGridView1.CurrentCell.ColumnIndex.Equals(2) && e.RowIndex != -1)
            {
                if (dataGridView1.CurrentCell != null && dataGridView1.CurrentCell.Value != null)
                {
                    if (MessageBox.Show("Do you want to save this image?", "Prompt", MessageBoxButtons.YesNo, MessageBoxIcon.Warning) == DialogResult.Yes)
                    {
                        try {
                            SaveFileDialog saveFileDialog1 = new SaveFileDialog();
                            //saveFileDialog1.InitialDirectory = @"C:\";      
                            saveFileDialog1.Title = "Save Image";
                            //saveFileDialog1.CheckFileExists = true;
                            saveFileDialog1.CheckPathExists = true;
                            saveFileDialog1.DefaultExt = "png";
                            saveFileDialog1.Filter = "Image (*.png)|*.png|All files (*.*)|*.*";
                            saveFileDialog1.FilterIndex = 0;
                            saveFileDialog1.RestoreDirectory = true;
                            if (saveFileDialog1.ShowDialog() == DialogResult.OK)
                            {
                                Image image = (Image)dataGridView1.CurrentCell.Value;
                                image.Save(saveFileDialog1.FileName);
                                MessageBox.Show("Successfully saved", "Information", MessageBoxButtons.OK, MessageBoxIcon.Information);
                            }
                        }
                        catch
                        {
                            MessageBox.Show("Failed","Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
                        }
                    }
                }
            }
        }
    }
}

File: Sample VisualBasic Application/Form1.vb
Public Class Form1
    Private Async Sub button1_Click(sender As Object, e As EventArgs) Handles button1.Click
        Dim result = Await DevNet.QImage.Query(Me.textBox1.Text, CInt(Me.numericUpDown1.Value))
        dataGridView1.DataSource = result
    End Sub

    Private Sub dataGridView1_CellContentDoubleClick(sender As Object, e As DataGridViewCellEventArgs) Handles dataGridView1.CellContentDoubleClick
        If dataGridView1.CurrentCell.ColumnIndex.Equals(2) AndAlso e.RowIndex <> -1 Then

            If dataGridView1.CurrentCell IsNot Nothing AndAlso dataGridView1.CurrentCell.Value IsNot Nothing Then
                'If MessageBox.Show("Do you want to save this image?", "Prompt", MessageBoxButtons.YesNo, MessageBoxIcon.Warning) = DialogResult.Yes Then

                '    Try
                '        Dim saveFileDialog1 As SaveFileDialog = New SaveFileDialog()
                '        saveFileDialog1.Title = "Save Image"
                '        saveFileDialog1.CheckPathExists = True
                '        saveFileDialog1.DefaultExt = "png"
                '        saveFileDialog1.Filter = "Image (*.png)|*.png|All files (*.*)|*.*"
                '        saveFileDialog1.FilterIndex = 0
                '        saveFileDialog1.RestoreDirectory = True

                '        If saveFileDialog1.ShowDialog() = DialogResult.OK Then
                '            Dim image As Image = CType(dataGridView1.CurrentCell.Value, Image)
                '            image.Save(saveFileDialog1.FileName)
                '            MessageBox.Show("Successfully saved", "Information", MessageBoxButtons.OK, MessageBoxIcon.Information)
                '        End If

                '    Catch
                '        MessageBox.Show("Failed", "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning)
                '    End Try
                'End If
                Dim image As Image = CType(dataGridView1.CurrentCell.Value, Image)
                Me.pictureBox1.Image = image
            End If
        End If
    End Sub
End Class
